src/main/scala/com/exasol/spark/rdd/ExasolRDD.scala

```diff
@@ -35,7 +36,25 @@ class ExasolRDD(
 ) extends RDD[Row](sc, Nil)
     with LazyLogging {
 
-  @transient lazy val mainExaConnection: EXAConnection = {
+  // scalastyle:off null
+  @transient private var mainConnection: EXAConnection = null
+  @transient private var mainStatement: Statement = null
+  @transient private var mainResultSet: EXAResultSet = null
+  // scalastyle:on
+
+  def closeMainResources(): Unit = {
+    if (mainConnection != null) {
+      mainConnection.close()
+    }
+    if (mainStatement != null) {
+      mainStatement.close()
+    }
+    if (mainResultSet != null) {
+      mainResultSet.close()
+    }
```

Comment:
should we close them in the reverse order? From RS then statement to main connection?